### ISSUE TITLE
server crashes when cors is added

### ISSUE THREAD

**Issue by zilahir** [franciscop/server#108]
**Describe the bug**
Passing `const cors` as an argument to `server` and sending a `post`method to the defined endpoint, throws `internal server error` and, `middleware` error on the `console`

**To Reproduce**
1. `npm i server`
2. `touch index.js`
3. Add this

```
// Include it and extract some methods for convenience
const server = require('server');
const { get, post } = server.router;

const cors = [
    ctx => header("Access-Control-Allow-Origin", "*"),
    ctx => header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept"),
    ctx => ctx.method.toLowerCase() === 'options' ? 200 : false
  ];

// Launch server with options and a couple of routes
server({ port: 8080 }, cors, [
  get('/', ctx => 'Hello world'),
  post('/demo', ctx => console.log(ctx.data))
]);
```

4` node index.js`
5. Send a post request to: `http://localhost:8080/demo`

**Expected behavior**
Some data to returns. Instead getting errors.

**Comment by franciscop** [comment]
Hey thanks for the report with the code, do you have the error output?

**Comment by SidneyNemzer** [comment]
That error is unrelated to CORS, it's complaining that the /demo route didn't return any data to the client

**Comment by franciscop** [comment]
I didn't notice that in my first read, but that seems to be the case. There are two issues though:

- The error is not clear enough. I think adding the path and method would be much more clear, like `The POST /demo request did not return anything`.
- I believe that example is right in the documentation.

**Comment by franciscop** [comment]
I released `1.0.19` which should make things a lot more clear:

```
Your middleware did not return anything for this request:

GET /abc.png

This normally happens when no route was matched or if the router did not reply with anything. Make sure to return something, even if it's a catch-all error.

Documentation for reply: https://serverjs.io/documentation/reply/
Relevant issue: https://github.com/franciscop/server/issues/118
```
